python/ray/_private/workers/default_worker.py

```diff
@@ -270,15 +271,34 @@
         worker_launched_time_ms=worker_launched_time_ms,
     )
 
+    stdout_fileno = sys.stdout.fileno()
+    stderr_fileno = sys.stderr.fileno()
+    # We also manually set sys.stdout and sys.stderr because that seems to
+    # have an effect on the output buffering. Without doing this, stdout
+    # and stderr are heavily buffered resulting in seemingly lost logging
+    # statements. We never want to close the stdout file descriptor, dup2 will
+    # close it when necessary and we don't want python's GC to close it.
+    sys.stdout = ray._private.utils.open_log(
+        stdout_fileno, unbuffered=True, closefd=False
+    )
+    sys.stderr = ray._private.utils.open_log(
+        stderr_fileno, unbuffered=True, closefd=False
+    )
+
     worker = ray._private.worker.global_worker
 
-    # Setup log file.
-    out_file, err_file = node.get_log_file_handles(
-        get_worker_log_file_name(args.worker_type)
-    )
-    configure_log_file(out_file, err_file)
-    worker.set_out_file(out_file)
-    worker.set_err_file(err_file)
+    # Setup log file if rotation not enabled.
+    if os.getenv("RAY_ROTATION_MAX_BYTES", None):
+        worker.set_file_rotation_enabled(True)
+    else:
+        out_filepath, err_filepath = node.get_log_file_names(
+            get_worker_log_file_name(args.worker_type),
+            unique=False,  # C++ core worker process already creates the file, should use a deterministic function to get the same file path.
+            create_out=True,
+            create_err=True,
+        )
+        worker.set_out_file(out_filepath)
+        worker.set_err_file(err_filepath)
```

Comment:
This can be moved out of `else` since we always have the outfile and errfile.

Reply:
updated